packages/astro/src/runtime/client/dev-overlay/plugins/audit/a11y.ts

```diff
@@ -352,13 +352,72 @@ export const a11y: AuditRuleWithSelector[] = [
 	},
 	{
 		code: 'a11y-missing-content',
-		title: 'Missing content on element important for accessibility',
-		message: 'Headings and anchors must have content to be accessible.',
+		title: 'Missing content',
+		message:
+			'Headings and anchors must have an accessible name, which can come from: inner text, aria-label, aria-labelledby, an img with alt property, or an svg with a tag <title></title>.',
 		selector: a11y_required_content.join(','),
 		match(element: HTMLElement) {
 			// innerText is used to ignore hidden text
 			const innerText = element.innerText.trim();
-			if (innerText === '') return true;
+			if (innerText !== '') return false;
+
+			// Check for aria-label
+			const ariaLabel = element.getAttribute('aria-label')?.trim();
+			if (ariaLabel && ariaLabel !== '') return false;
+
+			// Check for valid aria-labelledby
+			const ariaLabelledby = element.getAttribute('aria-labelledby')?.trim();
+			if (ariaLabelledby) {
+				const ids = ariaLabelledby.split(' ');
+				for (const id of ids) {
+					const referencedElement = document.getElementById(id);
+					if (referencedElement && referencedElement.innerText.trim() !== '') return false;
+				}
+			}
+
+			// Check for <img> with valid alt attribute
+			const imgElements = element.querySelectorAll('img');
```

Comment:
`match` is a callback for a single `HTMLElement`, so this will run for every `img` on the page. Shouldn't this check if `element.localName === 'img'` and not use `querySelectorAll`?

Reply:
I think it just gets the imgs that are descendants of the element, because if you have multiple images inside the `<a>` only 1 of them needs to have an alt property

Reply:
but I guess all imgs should have alt property but I don't know if that's the scope of this rule?

Reply:
> `match` is a callback for a single `HTMLElement`, so this will run for every `img` on the page.

This rule is only applied to anchors and headings, so not an issue atm.

> Shouldn't this check if `element.localName === 'img'` and not use `querySelectorAll`?

It needs to inspect child elements to work for e.g. links with svgs or images only.

Reply:
Yeah, this looks fine to me!